src/Onnx/Session.cc

```diff
@@ -34,6 +46,36 @@ Session::Session(Core::Configuration const& config)
     Ort::SessionOptions session_opts;
     session_opts.SetIntraOpNumThreads(intraOpNumThreads_);
     session_opts.SetInterOpNumThreads(interOpNumThreads_);
+
+    auto providers = Ort::GetAvailableProviders();
+    switch (paramExecutionProviderType(config)) {
+        case ExecutionProviderType::cpu: {
+            if (std::find(providers.begin(), providers.end(), "CPUExecutionProvider") == providers.end()) {
+                error() << "Requested CPU execution provider for ONNX session but it is not available.";
+            }
+            break;
+        }
+        case ExecutionProviderType::cuda: {
+            if (std::find(providers.begin(), providers.end(), "CUDAExecutionProvider") == providers.end()) {
+                error() << "Requested CUDA execution provider for ONNX session but it is not available.";
+            }
+#ifndef MODULE_CUDA
+            error() << "Requested CUDA execution provider but RASR was not compiled with MODULE_CUDA which is required for it.";
+#endif
+            int deviceCount = 0;
+            if (cudaGetDeviceCount(&deviceCount) != cudaSuccess or deviceCount == 0) {
+                error() << "Requested CUDA execution provider but no CUDA device was found.";
+            }
+            OrtCUDAProviderOptionsV2* cuda_opts = nullptr;
+            Ort::ThrowOnError(Ort::GetApi().CreateCUDAProviderOptions(&cuda_opts));
+            session_opts.AppendExecutionProvider_CUDA_V2(*cuda_opts);
+            Ort::GetApi().ReleaseCUDAProviderOptions(cuda_opts);
+            break;
```

Comment:
```suggestion
#ifdef MODULE_CUDA
            int deviceCount = 0;
            if (cudaGetDeviceCount(&deviceCount) != cudaSuccess or deviceCount == 0) {
                error() << "Requested CUDA execution provider but no CUDA device was found.";
            }
            OrtCUDAProviderOptionsV2* cuda_opts = nullptr;
            Ort::ThrowOnError(Ort::GetApi().CreateCUDAProviderOptions(&cuda_opts));
            session_opts.AppendExecutionProvider_CUDA_V2(*cuda_opts);
            Ort::GetApi().ReleaseCUDAProviderOptions(cuda_opts);
            break;
#endif
```
I had to add this because I ran into the errors "‘cudaGetDeviceCount’ was not declared in this scope" and 
"‘cudaSuccess’ was not declared in this scope" when trying to compile with MODULE_CUDA disabled.